mlrun/model_monitoring/db/stores/v3io_kv/kv_store.py

```diff
@@ -285,7 +285,24 @@ def list_model_endpoints(
             endpoint = self.get_model_endpoint(
                 endpoint_id=endpoint_id,
             )
-            endpoint_list.append(endpoint)
+            label_in = True
+            if labels:
+                endpoint_labels = json.loads(endpoint["labels"])
+                for label in labels:
+                    if "=" in label:
+                        label, value = list(
+                            map(lambda x: x.strip(), label.split("=", 1))
+                        )
+                    else:
+                        value = None
+                    label_in = label in endpoint_labels and (
+                        not value or str(endpoint_labels[label]) == value
+                    )
+                    if not label_in:
+                        break
+
+            if label_in:
+                endpoint_list.append(endpoint)
```

Comment:
I just realized this is the same logic as in `SQLStoreBase`. Can we eliminate this duplication?

Reply:
Nice catch!